docs/how-to/cooperative_groups.rst

```diff
@@ -0,0 +1,412 @@
+
+.. code-block:: C++
+
+    void* params[] = {&d_vector, &d_block_reduced, &d_partition_reduced};
+    // Launching kernel from host.
+    HIP_CHECK(hipLaunchCooperativeKernel(vector_reduce_kernel<partition_size>,
+                                         dim3(num_blocks),
+                                         dim3(threads_per_block),
+                                         params,
+                                         0,
+                                         hipStreamDefault));
+
+
+3. The device side synchronization on single GPU:
+
+.. code-block:: C++
+
+  grid_group grid = this_grid();
+  grid.sync();
+
+Multi-Grid Synchronization
+-----------------------------
+
+This section describes the necessary step to be able to synchronize group over multiple GPUs:
+
+1. Check the cooperative launch capability over multiple GPUs:
+
+.. code-block:: C++
+  
+  // Check support of cooperative groups
+  std::vector<int> deviceIDs;
+  for(int deviceID = 0; deviceID < device_count; deviceID++) {
+  #ifdef __HIP_PLATFORM_AMD__
+    int supports_coop_launch = 0;
+    HIP_CHECK(
+      hipDeviceGetAttribute(
+        &supports_coop_launch, 
+        hipDeviceAttributeCooperativeMultiDeviceLaunch, 
+        deviceID));
+    if(!supports_coop_launch) {
+      std::cout << "Skipping, device " << deviceID << " does not support cooperative groups"
+                << std::endl;
+    }
+    else
+  #endif
+    {
+      std::cout << deviceID << std::endl;
+      // Collect valid deviceIDs.
+      deviceIDs.push_back(deviceID);
+    }
+  }
+
+  if(!deviceIDs.size())
+  {
+      std::cout << "No valid GPU found." << std::endl;
+  } else {
+        std::cout << "Valid GPUs number:" << deviceIDs.size() << std::endl;
+  }
+
+2. Launch the cooperative kernel over multiple GPUs:
+
+.. code-block:: C++
+
+    hipLaunchParams *launchParamsList = (hipLaunchParams*)malloc(sizeof(hipLaunchParams) * deviceIDs.size());
+    for(int deviceID : deviceIDs) {
+		// Set device
+		HIP_CHECK(hipSetDevice(deviceID));
+		
+		// Create stream
+        hipStream_t stream;
+        HIP_CHECK(hipStreamCreate(&stream));
+		
+		// Parameters
+		void* params[] = {&(d_vector[deviceID]), &(d_block_reduced[deviceID]), &(d_partition_reduced[deviceID])};
+		
+		// Set launchParams
+        launchParamsList[deviceID].func = (void*)vector_reduce_kernel<partition_size>;
+        launchParamsList[deviceID].gridDim = dim3(1);
+        launchParamsList[deviceID].blockDim = dim3(threads_per_block);
+        launchParamsList[deviceID].sharedMem = 0;
+        launchParamsList[deviceID].stream = stream;
+        launchParamsList[deviceID].args = params;
+    }
+
+    HIP_CHECK(hipLaunchCooperativeKernelMultiDevice(launchParamsList,
+                                                    (int)deviceIDs.size(),
+                                                    hipCooperativeLaunchMultiDeviceNoPreSync));
+
+3. The device side synchronization over multiple GPUs:
```

Comment:
Is this a procedure - what is this supposed to mean?

Reply:
This is actual synchronization of the threads. This section describe the necessary steps to be able to synchronize with the different types of groups.